tool.c

```diff
@@ -171,6 +171,72 @@ commands(int argc, char *argv[])
 	return ret;
 }
 
+static void
+graphviznode(struct node *n)
+{
+	size_t i;
+	char *order_only;
+	struct edge *e;
+
+	printf("\"%p\" [label=\"%s\"]\n", (void *)n, n->path->s);
+
+	e = n->gen;
+	if (!e || (e->flags & FLAG_WORK))
+		return;
+	e->flags |= FLAG_WORK;
+
+	for (i = 0; i < e->nin; i++)
+		graphviznode(e->in[i]);
+
+	if (e->nin == 1 && e->nout == 1) {
+		printf("\"%p\" -> \"%p\" [label=\" %s\"]\n", (void *)e->in[0], (void *)e->out[0], e->rule->name);
```

Comment:
Do you know the reason for the leading space? Ninja says it's for aesthetics, but it seems to look fine to me without it.

Reply:
I don't, I was following ninja, looks fine without to me too. Removed it for now.